harness/simdUtilities.js

```diff
@@ -0,0 +1,470 @@
+  float64x2.boolType = bool64x2;
+
+  float32x4.fromBits.push(float64x2);
+  int32x4.fromBits.push(float64x2);
+  int16x8.fromBits.push(float64x2);
+  int8x16.fromBits.push(float64x2);
+  uint32x4.fromBits.push(float64x2);
+  uint16x8.fromBits.push(float64x2);
+  uint8x16.fromBits.push(float64x2);
+
+  float64x2.fromBits = [float32x4, int32x4, int16x8, int8x16,
+  uint32x4, uint16x8, uint8x16];
+
+  int32x4.fromBits = [float32x4, int16x8, int8x16, uint32x4,
+  uint16x8, uint8x16];
+  int16x8.fromBits = [float32x4, int32x4, int8x16, uint32x4,
+  uint16x8, uint8x16];
+  int8x16.fromBits = [float32x4, int32x4, int16x8, uint32x4,
+  uint16x8, uint8x16];
+  uint32x4.fromBits = [float32x4, int32x4, int16x8, int8x16,
+  uint16x8, uint8x16];
+  uint16x8.fromBits = [float32x4, int32x4, int16x8, int8x16,
+  uint32x4, uint8x16];
+  uint8x16.fromBits = [float32x4, int32x4, int16x8, int8x16,
+  uint32x4, uint16x8];
+
+  simdTypes.push(float64x2);
+  simdTypes.push(bool64x2);
+}
+
+// SIMD utility functions.
+
+// Create a value for testing, with vanilla lane values, i.e. [0, 1, 2, ..]
+// for numeric types, [false, true, true, ..] for boolean types. These test
+// values shouldn't contain NaNs or other "interesting" values.
+function createTestValue(type) {
+  var lanes = [];
+  for (var i = 0; i < type.lanes; i++)
+    lanes.push(i);
+  return type.fn.apply(type.fn, lanes);
+}
+
+function createSplatValue(type, v) {
+  var lanes = [];
+  for (var i = 0; i < type.lanes; i++)
+    lanes.push(v);
+  return type.fn.apply(type.fn, lanes);
+}
+
+// SIMD reference functions.
+
+// Returns converted array buffer value of specified type.
+function simdConvert(type, value) {
+  if (type.buffer === undefined) return !!value;  // bool types
+  type.buffer[0] = value;
+  return type.buffer[0];
+}
+
+function checkValue(type, a, expect) {
+  var fail = false;
+  for (var i = 0; i < type.lanes; i++) {
+    var v = type.fn.extractLane(a, i);
+    var ev = simdConvert(type, expect(i));
+    if (!sameValue(ev, v) && Math.abs(ev - v) >= 0.00001)
+      fail = true;
+  }
+  if (fail) {
+    var lanes = [];
+    for (var i = 0; i < type.lanes; i++){
+      lanes.push(simdConvert(type, expect(i)));
+    }
+    $ERROR("expected SIMD." + type.name + "(" + lanes +
+        ") but found " + a.toString());
+  }
+}
+
+// SIMD reference functions.
+
+// Reference implementation of toString.
+function simdToString(type, value) {
+  value = type.fn.check(value);
+  var str = "SIMD." + type.name + "(";
+  str += type.fn.extractLane(value, 0);
+  for (var i = 1; i < type.lanes; i++) {
+    str += ", " + type.fn.extractLane(value, i);
+  }
+  return str + ")";
+}
+
+// Reference implementation of toLocaleString.
+function simdToLocaleString(type, value) {
+  value = type.fn.check(value);
+  var str = "SIMD." + type.name + "(";
+  str += type.fn.extractLane(value, 0).toLocaleString();
+  for (var i = 1; i < type.lanes; i++) {
+    str += ", " + type.fn.extractLane(value, i).toLocaleString();
```

Comment:
`", "` should be `","` (remove the trailing white space), cf. [SIMDConstructor.prototype.toLocaleString, step 3](http://tc39.github.io/ecmascript_simd/#simd-prototype-tolocalestring).